### PR TITLE
CHDEXCHAE-55 add support for S3 Filestore plugin

### PR DESCRIPTION
**Pull request by ThrawnCA**

### CODE REVIEW

File: ckanext/validation/utils.py [hunk 1]
```diff
@@ -138,6 +138,31 @@ def run_sync_validation(resource_data):
...
+        path = upload.get_path(resource_data['id'])
+    else:
+        try:
+            from ckanext.s3filestore.uploader import S3ResourceUploader
+        except Exception:
+            return path
+
+        if isinstance(upload, S3ResourceUploader):
+            filename = os.path.basename(resource_data["url"])
+            key_path = upload.get_path(resource_data["id"], filename)
```
Comment (duttonw) [inline ckanext/validation/utils.py:157]: Don't we have a 'download' api in Iuploader? Can't we use that if it exists?

Also to confirm this is a dataset resource not a static asset upload.

Comment (ThrawnCA) [reply ckanext/validation/utils.py:157]: The download API provides a response suitable for passing to Flask, typically a stream. However, what we need here is a URL to pass to Goodtables.

**Review by duttonw — APPROVED** [review]

File: ckanext/validation/utils.py (hunk 1 above)
Comment (ThrawnCA) [reply ckanext/validation/utils.py:157]: Actually, a better option might be to use the standard resource download URL, `/resource/<UUID>/download`. I'm not sure why it was necessary to pass the direct S3 URL to Goodtables.

Comment (duttonw) [reply ckanext/validation/utils.py:157]: does it need auth, how does it with for private resources?

Comment (ThrawnCA) [reply ckanext/validation/utils.py:157]: Private resources will likely succeed, since they'll get a signed S3 URL.

**Pull request merged**